index-scheduler/src/autobatcher.rs

```diff
@@ -173,11 +220,51 @@ impl BatchKind {
             (this, kind) if !index_already_exists && this.allow_index_creation() == Some(false) && kind.allow_index_creation() == Some(true) => {
                 Break(this)
             },
+            //
+            // 1. If both task don't interact with primary key -> we can continue
+            // 2. Else ->
+            //   2.1 If we already have a primary-key ->
+            //     2.1.1 If the task we're trying to accumulate have a pk -> it must be equal to our primary key to continue
+            //     2.1.2 If the task don't have a primary-key -> we can continue
+            //   (We've already ensured that the current batch was correct according to our pk in a previous step of the autobatcher)
+            //   2.2 If we don't have a primary-key ->
+            //     2.2.1 If both the batch and the task have a primary key they should be equal
+            //     2.2.2 If the batch is set to Some(None), the task should be too
+            //     2.2.3 If the batch is set to None -> we can continue
+            //
```

Comment:
We can maybe remove that part as it is already below. I don't like to duplicate applications as we must keep both in think and we will fail to do so.
```suggestion
```

Reply:
True, it helped me during the implementation, but the in-code comments should be enough. Thanks for the suggestion :+1: